test/suites/subnet/vnet_subnet_id_test.go

```diff
@@ -0,0 +1,121 @@
+		nodes := env.EventuallyExpectCreatedNodeCount("==", 1)
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+
+		vm := env.GetVM(nodes[0].Name)
+		Expect(vm.Properties).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces).To(HaveLen(1))
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces[0].ID).ToNot(BeNil())
+		nicID, err := arm.ParseResourceID(*vm.Properties.NetworkProfile.NetworkInterfaces[0].ID)
+		Expect(err).ToNot(HaveOccurred())
+
+		// The NIC should have the right subnet
+		nic := env.GetNetworkInterface(nicID.Name)
+		Expect(nic.Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations).To(HaveLen(1))
+		Expect(nic.Properties.IPConfigurations[0].Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet.ID).To(Equal(subnet.ID))
+
+		// The NIC should have the right NSG
+		Expect(nic.Properties.NetworkSecurityGroup).ToNot(BeNil())
+		Expect(nic.Properties.NetworkSecurityGroup.ID).ToNot(BeNil())
+		Expect(*nic.Properties.NetworkSecurityGroup.ID).To(MatchRegexp(`aks-agentpool-\d{8}-nsg`))
+	})
+	It("should reject the AKSNodeClass if the subnet ID is invalid", func() {
+		nodeClass.Spec.VNETSubnetID = lo.ToPtr("/subnets/fake-subnet")
+		err := env.Client.Create(env.Context, nodeClass)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("Invalid value"))
+		Expect(err.Error()).To(ContainSubstring("spec.vnetSubnetID"))
+		Expect(err.Error()).To(ContainSubstring("should match"))
+	})
+
+	It("should mark the AKSNodeClass as unready if the subnet is NotFound and all back to a different nodeclass", func() {
+		newNodeClass := env.DefaultAKSNodeClass()
+		newNodepool := env.DefaultNodePool(newNodeClass)
+		newNodepool.Spec.Weight = lo.ToPtr(int32(10))
+		newNodeClass.Spec.VNETSubnetID = lo.ToPtr("/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/sillygeese/providers/Microsoft.Network/virtualNetworks/karpenter/subnets/nodeclassSubnet2")
+		env.ExpectCreated(nodeClass, nodePool, newNodeClass, newNodepool, dep)
+
+		By("falling back to original nodeclass due to misconfigured subnet")
+		env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+		node := env.EventuallyExpectCreatedNodeCount("==", 1)[0]
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+		Expect(node.Labels[karpv1.NodePoolLabelKey]).To(Equal(nodePool.Name)) // Validate we are are falling back to the original nodepool
+
+		// Verify the new nodeclass with full subnet has SubnetReady condition set to false
+		Eventually(func(g Gomega) {
+			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(newNodeClass), newNodeClass)).To(Succeed())
+			condition := newNodeClass.StatusConditions().Get(v1beta1.ConditionTypeSubnetReady)
+			g.Expect(condition.IsFalse()).To(BeTrue())
+		}).Should(Succeed())
+	})
```

Comment:
I will be adding many more tests here for followup validation, so figured it should be its own suite